packages/dialect-react-ui/components/Avatar/index.tsx

```diff
@@ -33,18 +35,25 @@ const CardinalAvatar = ({
   placeholder?: React.ReactNode;
   className?: string;
 }) => {
-  const { addressImage, loadingImage } = useAddressImage(connection, address);
+  const reverseEntry = useSWR([connection, address], getReverseEntry);
+  const displayName = formatName(
+    reverseEntry.data?.parsed.namespaces,
+    reverseEntry.data?.parsed.entryName
+  );
+  const [_namespace, handle] = displayName ? breakName(displayName) : [];
+  const addressImage = useSWR(handle, tryGetImageUrl);
```

Comment:
handle may not be specific enough, given it's global name spacing. maybe `/twitter-image-addresses/${handle}` or something